### PR TITLE
[INFRA-2953] Switch packer templates to HCL language

### PR DESCRIPTION
**Pull request by dduportal**
This PR switches the packer templates of the repository from JSON to HCL, to provide a modern Packer usage.

It's a foundation to merge most of the templates and avoid code reuse, provides comments to help contributions, and a road to remove code (less shell, no more need of the `docker-packer` image).

Details and intent in https://issues.jenkins.io/browse/INFRA-2953

### CODE REVIEW

**Review by timja — APPROVED** [review]
if it works

**Comment by slide** [comment]
What are the advantages of using HCL over JSON?

**Comment by dduportal** [comment]
@slide https://issues.jenkins.io/browse/INFRA-2953

**Comment by slide** [comment]
I missed the title change mentioning the Jira, thanks for helping me see it :-)

**Comment by dduportal** [comment]
@slide No problem ;) Feel free to drop in on IRC if you want more details to avoid polluting the PR ;) I'm available to answer questions / be challenged

**Comment by timja** [comment]
> @slide [issues.jenkins.io/browse/INFRA-2953](https://issues.jenkins.io/browse/INFRA-2953) (I'm trying to write down intents before a PR)

better to put that info in the PR imo, the jira doesn't really add any value unless it's capturing work for later.

**Review by olblak — APPROVED** [review]
I made a small comment but nothing that prevent this PR to move forward

File: aws/ubuntu-18-agent.amd64.pkr.hcl [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  type    = string
+}
+
+variable "compose_version" {
+  type    = string
+}
+
+variable "maven_version" {
+  type    = string
+}
```
[the remark this answers is not in the document]
Comment (dduportal) [reply aws/ubuntu-18-agent.amd64.pkr.hcl:23]: I was thinking to add a https://www.packer.io/docs/templates/hcl_templates/variables#variable-definitions-pkrvars-hcl-and-auto-pkrvars-hcl-files in another PR: we are in sync 👍

**Pull request merged**